examples/common/mqtt_agent/mqtt_agent_task.c

```diff
@@ -506,26 +506,51 @@ static MQTTStatus_t prvCreateMQTTConnection( bool xIsReconnect )
 
 static BaseType_t prvCreateTLSConnection( NetworkContext_t * pxNetworkContext )
 {
-    BaseType_t xConnected = pdFAIL;
+    BaseType_t xConnected = pdPASS;
 
     TlsTransportStatus_t xNetworkStatus = TLS_TRANSPORT_CONNECT_FAILURE;
     NetworkCredentials_t xNetworkCredentials = { 0 };
 
-#ifdef democonfigUSE_AWS_IOT_CORE_BROKER
+#if defined( democonfigUSE_AWS_IOT_CORE_BROKER )
+    static const char * ppcAlpnProtocols[ 2 ] = { NULL, NULL };
 
-    /* ALPN protocols must be a NULL-terminated list of strings. Therefore,
-     * the first entry will contain the actual ALPN protocol string while the
-     * second entry must remain NULL. */
-    const char * pcAlpnProtocols[] = { NULL, NULL };
+    /*
+     * When connecting AWS IoT Core via mqtt on port 443, an alpn is required.
+     * For more information, refer to the AWS IoT Core Device documentation at:
+     * https://docs.aws.amazon.com/iot/latest/developerguide/protocols.html
+     */
+    if( ulBrokerPort == 443U )
+    {
+#if defined( democonfigCLIENT_USERNAME )
```

Comment:
Would it be more user friendly to define a config such as `democonfigUSE_CUSTOM_AUTH` ?
`democonfigCLIENT_USERNAME` could also be used for other  use cases like metrics reporting.

Reply:
democonfigCLIENT_USERNAME is better if targeting non-AWS mqtt services.